bintool/bintool.cpp

```diff
@@ -961,7 +1019,7 @@ int encrypt(elf_file *elf, block *new_block, const private_t aes_key, const publ
 }
 
 
-std::vector<uint8_t> encrypt(std::vector<uint8_t> bin, uint32_t storage_addr, uint32_t runtime_addr, block *new_block, const private_t aes_key, const public_t public_key, const private_t private_key, bool hash_value, bool sign) {
+std::vector<uint8_t> encrypt(std::vector<uint8_t> bin, uint32_t storage_addr, uint32_t runtime_addr, block *new_block, const aes_key_t aes_key, const public_t public_key, const private_t private_key, std::vector<uint8_t> iv_salt, bool hash_value, bool sign) {
```

Comment:
Why does this `encrypt` function return a `std::vector<uint8_t>`, whilst the `encrypt` function with a similar-looking signature on line 928 just returns an `int` ?

Reply:
The BIN versions of most functions returns a vector with the encrypted/signed/etc data, but the ELF versions modify the ELF file in place so just return ints.

Reply:
Aha, thanks :slightly_smiling_face: